Issue #3497: Round DECIMAL casts

hawkfish opened the pull request:
Use rounding for DECIMAL => INTEGER casts instead of truncation.

Commit:
Issue #3497: Round DECIMAL casts

Use rounding for DECIMAL => INTEGER casts instead of truncation.

Commit:
Issue #3497: Round DECIMAL casts

Tweak TPC-H result for new rounding.

Commit:
Issue #3497: Round DECIMAL casts

Updates some small test changes from the rounding.

Commit:
Issue #3497: Round DECIMAL casts

Update ODBC results.

Mytherin reviewed (approved):
Thanks for the PR! Looks good to me. Could you just have a look at the failing CI?

Commit:
Merge branch 'master' of github.com:duckdb/duckdb

Commit:
Merge branch 'master' into hawkfish-cast-round

Commit:
Merge branch 'master' into hawkfish-cast-round

Commit:
Merge branch 'master' into hawkfish-cast-round

Commit:
Merge branch 'master' into hawkfish-cast-round

Commit:
Merge branch 'hawkfish-cast-round' of github.com:hawkfish/duckdb into hawkfish-cast-round

hawkfish commented:
Part of the problem here is that the test comparison code is casting the expected "0.5" to 0 (because the result type is `BIGINT`) which then matches the previous truncated `SUM(CAST(0.5 AS SMALLINT))`. The new rounding logic produces a `1`, which no longer matches the way that the test code casts 0.5.

We allow casting '0.5' to an `INTEGER` in lenient mode, so I will make that consistent with the other casts. This will not fix the binding issue, but it will fix the failure.

Commit:
Issue #3497: Round DECIMAL casts

Use rounding for lenient VARCHAR => INTEGER casts instead of truncation.
This makes all integer rounding consistent.

hawkfish commented:
There are more ODBC expected that need to be updated, and I've made a PR for the `psqlodbc` repository. When that is available, I can updates the workflow files.

Commit:
Issue #3497: Round DECIMAL casts

Point at temporary ODBC results.

Commit:
Issue #3497: Round DECIMAL casts

Point at newer ODBC rounding results.
